src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,56 +1,128 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface Props {
+  onAdd: (movieToAdd: Movie) => void;
+}
+
+const initialFormState = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+export const NewMovie: React.FC <Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [newMovieInfo, setNewMovieInfo] = useState(initialFormState);
+
+  const {
+    title,
+    description,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  } = newMovieInfo;
+
+  const handleChange = (formField: string | number, value: string | number) => {
+    setNewMovieInfo((pevMovieInfo) => ({
+      ...pevMovieInfo,
+      [formField]: value,
+    }));
+  };
+
+  const clearForm = () => {
+    setNewMovieInfo(initialFormState);
+  };
+
+  const trimTextField = (textField: string): string => textField.trim();
+
+  const isAllRequiredFieldsFilled = () => {
+    return (
+      trimTextField(title)
+      && trimTextField(imgUrl)
+      && trimTextField(imdbUrl)
+      && trimTextField(imdbId)
+    );
+  };
```

Comment:
we call this function on every render so let's make it const
```suggestion
  const isAllRequiredFieldsFilled =
      trimTextField(title)
      && trimTextField(imgUrl)
      && trimTextField(imdbUrl)
      && trimTextField(imdbId)
  };
```